src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/consumer/impl/MessageConsumerImpl.java

```diff
@@ -46,7 +51,19 @@ void init() {
 
     @PreDestroy
     void close() {
-        consumer.close();
+        synchronized(lock) {
+            try {
```

Comment:
for testing, see this example - https://www.baeldung.com/java-synchronized#1-synchronized-instance-methods
we can run a large amount of threads (1000?) and have them all call close() at he same time, and verify the consumer.close() gets called the same amount of times (1000?)

Reply:
Added this to the test.